### PR TITLE
Add windows server core 2004 helix image

### PR DESCRIPTION
**Pull request by safern**
cc: @jaredpar @danmosemsft @MattGal

### CODE REVIEW

**Review by MattGal — APPROVED** [review]
LGTM

File: src/windowsservercore/2004/helix/amd64/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# escape=`
+FROM mcr.microsoft.com/windows/servercore:2004-amd64
+
+SHELL ["cmd", "/S", "/C"]
```
Comment (MichaelSimons) [inline src/windowsservercore/2004/helix/amd64/Dockerfile:4]: (NIT) This isn't necessary as it is the default shell.

File: src/windowsservercore/2004/helix/amd64/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+RUN curl -SL --output %TEMP%\python.zip https://www.nuget.org/api/v2/package/python/3.7.3 `
+    && md C:\Python C:\PythonTemp `
+    && tar -zxf %TEMP%\python.zip -C C:\PythonTemp `
+    && xcopy /s c:\PythonTemp\tools C:\Python `
+    && rd /s /q c:\PythonTemp `
+    && del /q %TEMP%\python.zip `
+    && setx /M PYTHONPATH "C:\Python\Lib;C:\Python\DLLs;"
+
+RUN md c:\\helixtmp && pushd c:\\helixtmp &&`
```
Comment (MichaelSimons) [inline src/windowsservercore/2004/helix/amd64/Dockerfile:15]: (INCONSISTENT FORMATTING) The previous section puts `&&` at the beginning of the line where this section puts it at the end.  The most common format used in Dockerfiles is to put it at the front of the line.

**Review by MichaelSimons — APPROVED** [review]
LGTM, I have a couple minor comments.

**Comment by mthalman** [comment]
@safern - I've updated this to add the necessary pipeline files for supporting Server Core 2004 builds.

**Comment by safern** [comment]
Thanks @mthalman I didn't know I had to add changes to have the image build.

File: src/windowsservercore/2004/helix/amd64/Dockerfile (hunk 1 above)
Comment (safern) [reply src/windowsservercore/2004/helix/amd64/Dockerfile]: done.

File: src/windowsservercore/2004/helix/amd64/Dockerfile (hunk 2 above)
Comment (safern) [reply src/windowsservercore/2004/helix/amd64/Dockerfile]: Done.

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**